scripts/prefect_scripts/bo_hitl_slack_tutorial.py

```diff
@@ -0,0 +1,314 @@
+        }
+        return parameters, self.trial_count
+        
+    def complete_trial(self, trial_index: int, raw_data: float):
+        """Mock trial completion using Service API pattern"""
+        self.completed_trials.append((trial_index, raw_data))
+        print(f"Completed trial {trial_index} with value {raw_data:.4f}")
+        
+    def get_best_parameters(self):
+        """Get best parameters so far"""
+        if not self.completed_trials:
+            return None
+        best_trial = min(self.completed_trials, key=lambda x: x[1])
+        return {"trial_index": best_trial[0], "value": best_trial[1]}
+
+
+def mock_branin(x1: float, x2: float) -> float:
+    """
+    Mock Branin function implementation
+    Global minimum at (π, 2.275) and (-π, 12.275) and (9.42478, 2.475) with value 0.397887
+    """
+    a = 1
+    b = 5.1 / (4 * math.pi**2)
+    c = 5 / math.pi
+    r = 6
+    s = 10
+    t = 1 / (8 * math.pi)
+    
+    return a * (x2 - b * x1**2 + c * x1 - r)**2 + s * (1 - t) * math.cos(x1) + s
+
+
+class ExperimentInput(RunInput):
+    """Input model for experiment evaluation"""
+    objective_value: float
+    notes: str = ""
+    
+    class Config:
+        description = "Please evaluate the suggested experiment and enter the objective value"
+
+
+@task
+def setup_ax_client() -> MockAxClient if not AX_AVAILABLE else AxClient:
+    """Initialize the Ax client with Branin function optimization setup using Service API"""
+    logger = get_run_logger()
+    
+    if AX_AVAILABLE:
+        ax_client = AxClient()
+        logger.info("Using real Ax Service API")
+    else:
+        ax_client = MockAxClient()
+        logger.info("Using mock Ax implementation")
+    
+    # Define the optimization problem for the Branin function using Service API pattern
+    ax_client.create_experiment(
+        name="branin_bo_hitl_experiment",
+        parameters=[
+            {
+                "name": "x1",
+                "type": "range", 
+                "bounds": [-5.0, 10.0],
+                "value_type": "float",
+            },
+            {
+                "name": "x2",
+                "type": "range",
+                "bounds": [0.0, 15.0],
+                "value_type": "float", 
+            },
+        ],
+        objectives={"branin": ObjectiveProperties(minimize=True)} if AX_AVAILABLE else {"branin": "minimize"}
+    )
+    
+    logger.info("Initialized Ax client for Branin function optimization")
+    return ax_client
+
+
+@task 
+def get_next_suggestion(ax_client) -> Tuple[Dict, int]:
+    """Get the next experiment suggestion from Ax using Service API"""
+    logger = get_run_logger()
+    
+    parameters, trial_index = ax_client.get_next_trial()
+    logger.info(f"Generated trial {trial_index}: x1={parameters['x1']:.3f}, x2={parameters['x2']:.3f}")
+    
+    return parameters, trial_index
+
+
+@task
+def complete_experiment(ax_client, trial_index: int, objective_value: float):
```

Comment:
I like the idea of this being a task, but this seems overly verbose in terms of just redefining the function. Is there not a simpler way to decorate the original function / register it as a task? E.g., using `task(...)` rather than `@task`? Refer to prefect docs and source code. Use perplexity MCP tool and GitHub MCP tool to search for examples with this construct.